llm_bench/python/utils/model_utils.py

```diff
@@ -30,14 +31,71 @@ def get_prompts(args):
                     with open(input_prompt, 'r', encoding='utf-8') as f:
                         for line in f:
                             data = json.loads(line)
-                            prompts_list.append(data['prompt'])
+                            if 'prompt' in data:
+                                if data['prompt'] != '':
+                                    prompts_list.append(data['prompt'])
+                                else:
+                                    raise RuntimeError('== prompt should not be empty string ==')
+                            else:
+                                raise RuntimeError('== key word "prompt" does not exist in prompt file ==')
                 else:
                     raise RuntimeError('== The prompt file does not exist ==')
             else:
-                prompts_list.append(input_prompt)
+                raise RuntimeError('== The prompt file should be ended with .jsonl ==')
     return prompts_list
 
 
+def init_image_param():
+    return {'prompt': '', 'width': 0, 'height': 0, 'steps': 0, 'guidance_scale': ''}
+
+
+def get_image_param_from_prompt_file(args):
+    image_param_list = []
+    image_param = init_image_param()
+    if args['prompt'] is None and args['prompt_file'] is None:
+        image_param['prompt'] = 'sailing ship in storm by Leonardo da Vinci'
+        image_param_list.append(image_param)
+    elif args['prompt'] is not None and args['prompt_file'] is not None:
+        raise RuntimeError('== prompt and prompt file should not exist together ==')
+    else:
+        if args['prompt'] is not None:
+            if args['prompt'] != '':
+                image_param['prompt'] = args['prompt']
+                image_param_list.append(image_param)
+            else:
+                raise RuntimeError('== prompt should not be empty string ==')
+        else:
+            input_prompt = args['prompt_file']
+            if input_prompt.endswith('.jsonl'):
+                if os.path.exists(input_prompt):
+                    log.info(f'Read prompts from {input_prompt}')
+                    with open(input_prompt, 'r', encoding='utf-8') as f:
```

Comment:
What will happen if it is not 'utf-8' file?

Reply:
1. For 'utf-16' prompt file will throw exception:
  File "/usr/lib/python3.10/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfe in position 0: invalid start byte
2.  For 'ascii' prompt file, params in prompt file can be read correctly.